plasma_framework/docs/integration-docs/integration-doc.md

```diff
@@ -565,18 +517,292 @@ PlasmaFramework.processExits([
+  [0x7876f3035bddd396f99ac58a5da145687e1dae2dff38d6c6e4fbdd76d548ed945ccc4b9e8226c87dbf8648cf09e8bd89926fdfb84e632b211e27a6a33f2882d01c]
+])
+```
+
+### Piggybacking on an in-flight exit input / output
+
+1. Get the amount of ETH to cover the bond for piggybacking in-flight exit as described [here](#in-flight-exit-bonds).
+
+2. To piggyback on in-flight exit input call:
+```
+PaymentExitGame.piggybackInFlightExitOnInput({
+  inFlightTx,
+  inputIndex
+})
+```
+To piggyback on in-flight exit output call:
+```
+PaymentExitGame.piggybackInFlightExitOnOutput({
+  inFlightTx,
+  outputIndex
+})
+```
+Appropriate amount of ETH needs to be provided to cover the bond.
+
+### Parameters
+
+#### inFlightTx (bytes)
+RLP encoded in-flight transaction.
+
+#### inputIndex (uint16)
+Index of the input to piggyback on.
+
+#### outputIndex (uint16)
+Index of the output to piggyback on.
+
+
+#### Examples:
+
+```
+PaymentExitGame.piggybackInFlightExitOnInput([
+  0xf87701e1a0000000000000000000000000000000000000000000000000000000003b9aca00f1f001ee947a809718aec76d8ac282a825be98e6ba4fc01fb89400000000000000000000000000000000000000008307a12080a00000000000000000000000000000000000000000000000000000000000000000,
+  0
+])
+```
+
+```
+PaymentExitGame.piggybackInFlightExitOnOutput([
+  0xf87701e1a0000000000000000000000000000000000000000000000000000000003b9aca00f1f001ee947a809718aec76d8ac282a825be98e6ba4fc01fb89400000000000000000000000000000000000000008307a12080a00000000000000000000000000000000000000000000000000000000000000000,
+  0
+])
+```
 
-**TODO:** Piggybacking on an In-flight Exit
+### Challenging canonicity of an in-flight exit
+
+To challenge canonicity of an in-flight exit show a competing transaction:
+```
+PaymentExitGame.challengeInFlightExitNotCanonical({
+    inputTx,
+    inputUtxosPos,
+    inFlightTx,
+    inFlightTxInputIndex,
+    competingTx,
+    competingTxInputIndex,
+    competingTxPos,
+    competingTxInclusionProof,
+    competingTxWitness
+})
+```
+
+### Parameters
 
-**TODO:** Challenging an In-flight Exit as non-canonical 
+#### inputTx (bytes)
+Transaction that created input shared by in-flight transaction and its competitor.
 
-**TODO:** Responding to an In-flight Exit non-canonical challenge 
+#### inFlightTx (bytes)
+RLP encoded in-flight transaction.
 
-**TODO:** Challenging an In-flight Exit input spent 
+#### inFlightTxInputIndex (uint16)
+Index of shared input in in-flight transaction.
 
-**TODO:** Challenging an In-flight Exit output spent 
+#### competingTx (bytes)
+RLP-encoded competing transaction that spends the shared input.
 
+#### competingTxInputIndex (uint16)
+Index of shared input in competing transaction
+
+#### competingTxPos (uint256)
+Transaction position of competing tx, which is utxo position of 0 indexed output.
```

Comment:
I find this part a confusing: `which is utxo position of 0 indexed output`
Can we just leave it out?